pkg/cache/v3/order_test.go

```diff
@@ -0,0 +1,49 @@
+package cache
+
+import (
+	"errors"
+	"sort"
+	"testing"
+
+	"github.com/envoyproxy/go-control-plane/pkg/resource/v3"
+)
+
+func TestOrderKeys(t *testing.T) {
+	unorderedKeys := keys{
+		{
+			ID:      1,
+			TypeURL: resource.EndpointType,
+		},
+		{
+			ID:      2,
+			TypeURL: resource.ClusterType,
+		},
+		{
+			ID:      4,
+			TypeURL: resource.ListenerType,
+		},
+		{
+			ID:      3,
+			TypeURL: resource.RouteType,
+		},
+		{
+			ID:      5,
+			TypeURL: resource.ScopedRouteType,
+		},
+	}
+
+	orderedKeys := unorderedKeys
+	sort.Sort(orderedKeys)
+
+	if !sort.IsSorted(orderedKeys) {
```

Comment:
```suggestion
assert.True(t, sort.IsSorted(orderedKeys))
assert.NotEqual(t, unorderedKeys, orderedKeys)
```

This shows that the sorting does something, then you can add a canonically sorted expected value as well.